lib/journaled/outbox/adapter.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Journaled
+  module Outbox
+    # Outbox-style delivery adapter for custom event processing
+    #
+    # This adapter stores events in a database table instead of enqueuing to ActiveJob.
+    # Events are processed by separate worker daemons that poll the database.
+    #
+    # Setup:
+    # 1. Generate migrations: rails generate journaled:database_events
+    # 2. Run migrations: rails db:migrate
+    # 3. Configure: Journaled.delivery_adapter = Journaled::Outbox::Adapter
+    # 4. Start workers: bundle exec rake journaled_worker:work
+    class Adapter < Journaled::DeliveryAdapter
+      class TableNotFoundError < StandardError; end
+
+      # Delivers events by inserting them into the database
+      #
+      # @param events [Array] Array of journaled events to deliver
+      # @param ** [Hash] Additional options (ignored, for interface compatibility)
+      # @return [void]
+      def self.deliver(events:, **)
+        check_table_exists!
+
+        records = events.map do |event|
+          # Exclude the application-level id - the database will generate its own using uuid_generate_v7()
+          event_data = event.journaled_attributes.except(:id)
```

Comment:
Confirmed that we're injecting the true PK ID into the payload later & that tests cover that, but I'm wondering what else might even care about the `id` prior to record insertion -- maybe we can set `id` to `nil` or remove it from the `Event` API surface entirely if Journaled is configured to use the outbox adapter... 🤔